tests/test_restrictions.py

```diff
@@ -0,0 +1,143 @@
+import pandas as pd
+from pandas.util.testing import assert_series_equal
+
+from zipline.finance.restrictions import (
+    RESTRICTION_STATES,
+    Restriction,
+    InMemoryRestrictions,
+    StaticRestrictions,
+    NoopRestrictions,
+)
+from zipline.testing.fixtures import (
+    WithDataPortal,
+    ZiplineTestCase,
+)
+
+str_to_ts = lambda dt_str: pd.Timestamp(dt_str, tz='UTC')
+
+
+class RestrictionsTestCase(WithDataPortal, ZiplineTestCase):
+
+    ASSET_FINDER_EQUITY_SIDS = 1, 2, 3
+
+    @classmethod
+    def init_class_fixtures(cls):
+        super(RestrictionsTestCase, cls).init_class_fixtures()
+        cls.ASSET1 = cls.asset_finder.retrieve_asset(1)
+        cls.ASSET2 = cls.asset_finder.retrieve_asset(2)
+        cls.ASSET3 = cls.asset_finder.retrieve_asset(3)
+
+    def test_in_memory_restrictions(self):
```

Comment:
Other cases that we should probably test:

- Multiple assets with restrictions on the same date shouldn't interact.  (Right now no assets ever have restrictions that apply on the same day.)
- The same asset with consecutive `FROZEN` or `ALLOWED` entries (this should either be an error, or we should ensure that it works.  I'm inclined to say the former, since there's a clear and obvious semantics for it.)
- Restrictions with intraday effective_dates.  (See my note below re: refactoring this to be more readable/parametrizable.
- Re-ordering of the restrictions here and asserting that there's no change in behavior.